Ocarina/OcarinaManager.swift

```diff
@@ -90,7 +92,13 @@ public class OcarinaManager: NSObject {
     fileprivate func dataTask(for url: URL) -> URLSessionDataTask {
         var request = URLRequest(url: url)
         request.cachePolicy = .reloadIgnoringLocalCacheData
-        request.setValue("Ocarinabot", forHTTPHeaderField: "User-agent")
+        
+        if (!self.userAgent.isEmpty) {
```

Comment:
If you make the userAgent optional it makes the code a lot cleaner. If userAgent is resolved to nil it will use `"Ocarinabot"` instead.

```suggestion
    let userAgent = self.userAgent ?? "Ocarinabot"
    request.setValue(userAgent, forHTTPHeaderField: "User-agent")
```